### ISSUE TITLE
Custom status flags: is there a safe value or range?

### ISSUE THREAD

**Issue by teppokoivula** [ryancramerdesign/ProcessWire#1518]
More a question than an issue:

For certain use cases I've used a custom status flag for pages. This is handy and efficient for searching, and seemed to fit perfectly with current implementations of hidden etc. Back then status codes between 32 and 512 were free, so I decided to use 128 for my own purposes, but as of 2.7 that's also reserved to core.

I'm currently planning to migrate my own stuff from old status code to something else, and thus the question in the topic: is there a "safe range" I can use, or should I ditch the idea of a custom status altogether and go with page field instead?

**Comment by ryancramerdesign** [comment]
I would say it's not safe to use status flags for custom needs. Best to use a custom field, whether an integer field or a Page field. I think that a good future addition to PW would be to support the option for a custom field to have its value stored or duplicated in the `pages` table for needs like you've mentioned, but that would need to be some field that is common to every template.

**Comment by teppokoivula** [comment]
Thanks! Going with a field has it's drawbacks and will require a bit more work at this point, but that's better than having to redo it every time these things change :)

Just for the record, in this case the extra flag was for "hidden for menu" status, which I've added to the list of statuses in the settings tab too. "Hidden" is rarely what the user *really* wanted, and after one too many questions about hidden pages not showing up in site search, I decided to add a custom status for this purpose alone.

Since it's the responsibility of front-end developer to actually implement the "hidden from menu" behaviour, this probably isn't a good candidate for core, even if it's essential for our sites/clients and has been asked a few times on the forum :)

Closing this issue, case solved.
